sopel/test_tools.py

```diff
@@ -65,11 +66,18 @@ def __init__(self, nick, admin=False, owner=False):
         self.config = MockConfig()
         self._init_config()
 
+        self.output = []
+
         if admin:
             self.config.core.admins = [self.nick]
         if owner:
             self.config.core.owner = self.nick
 
+    def _store(self, string, *args, **kwargs):
+        self.output.append(string.strip())
+
+    write = msg = say = notice = action = reply = _store
```

Comment:
I wonder which of these functions are actually high-level or low-leve methods. That means: which one use the `send` method, and which one are just fancy shortcut over a low-level method (which uses `send`)?

Basically, I'm wondering which of these methods really requires the `send` method.

Reply:
```
+----------------+    +--------------+    +-------------+
|  Sopel.action  +--->+  Sopel.say   +--->+             |
|  Sopel.msg     |    +--------------+    |             |    +---------------+    +--------------------------+
|  Sopel.reply   |                        | Sopel.write +--->+ irc.Bot.write +--->+ asynchat.async_chat.send |
+----------------+    +--------------+    |             |    +---------------+    +--------------------------+
                      | Sopel.notice +--->+             |
                      +--------------+    +-------------+
```
So ultimately, the bot's `write` method calls `send`. Since `MockSopel` doesn't subclass `Sopel`, all the other methods need to be defined (and point to `_store` in the mean time). Ideally, `MockSopel` will inherit the actual shortcut methods from `Sopel` so that any additional processing that happens
(e.g., in `reply` or even `say`) is actually executed.

This is what I imagined when I said: https://github.com/sopel-irc/sopel/pull/1542#issuecomment-481223596

Do you have a preference for how `write = msg = say = notice = action = reply = _store` should be organized differently? For example, more explicitly, like:
```python
def _store(self, string, *args, **kwargs):
    ...

write = _store
say = write
notice = write
msg = reply = action = say
```
to more directly represent how the methods are routed in `Sopel`?

Reply:
This is fine as-is. Big up for the lovely graph you made, that helped me a lot understanding the issue. Thank you!